flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/python/PythonTableFunction.java

```diff
@@ -48,6 +48,7 @@
 	private final PythonFunctionKind pythonFunctionKind;
 	private final boolean deterministic;
 	private final PythonEnv pythonEnv;
+	private final boolean usedInRowBasedOperation;
```

Comment:
I guess we could also add row-based support in join_lateral/left_outer_join_lateral? I'm fine to do it in a separate PR if it you want.

Reply:
Yes. I have created https://issues.apache.org/jira/browse/FLINK-20712 to support this feature.